spec/cyanobyte.schema.json

```diff
@@ -0,0 +1,216 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "http://github.com/cyanobyte/cyanobyte.schema.json",
+  "title": "CyanoByte Specification Validator",
+  "description": "This file is used to validate whether a given document meets the CyanoByte specification.",
+  "type": "object",
+  "properties": {
+    "cyanobyte": {
+      "description": "The version of the CyanoByte specification this file follows.",
+      "type": "string"
+    },
+    "info": {
+      "description": "Info block of the file",
+      "type": "object",
+      "properties": {
+        "title": {
+          "description": "The title of the document. This should be used as the class name in the auto-generated code and therefore should be short (e.g. the part name).",
+          "type": "string"
+        },
+        "description": {
+          "description": "A description of the document.",
+          "type": "string"
+        },
+        "contact": {
+          "description": "Contact info for the maintainer.",
+          "type": "object",
+          "properties": {
+            "name": {
+              "description": "Name of the maintainer.",
+              "type": "string"
+            },
+            "url": {
+              "description": "URL of the maintainer organization.",
+              "type": "string",
+              "format": "uri"
+            },
+            "email": {
+              "description": "Email address for the maintainer.",
+              "type": "string",
+              "format": "email"
+            }
+          },
+          "required": [
+            "name",
+            "url",
+            "email"
+          ]
+        },
+        "license": {
+          "description": "License this document uses. TODO: Add validation other than it being a string.",
+          "type": "string"
+        },
+        "version": {
+          "description": "Version number for this document using kind of SymVer.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "title",
+        "description",
+        "contact",
+        "license",
+        "version"
+      ]
+    },
+    "i2c": {
+      "description": "I2C settings",
+      "type": "object",
+      "properties": {
+        "addressType": {
+          "description": "Addressing type of the component.",
+          "type": "string",
+          "enum": [
+            "7-bit",
```

Comment:
8-bit?

Reply:
I was thinking about it again, and if we support 8-bit, we need to add both a read and write address. These two addresses can be represented as one 7-bit address.